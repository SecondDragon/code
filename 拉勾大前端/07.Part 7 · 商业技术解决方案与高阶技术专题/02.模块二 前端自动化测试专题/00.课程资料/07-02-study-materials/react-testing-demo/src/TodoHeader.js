export default function TodoHeader () {
  return (
    <header className="header">
      <h1 title="todos">todos</h1>
      <input className="new-todo" placeholder="What needs to be done?" autoFocus data-testid="new-todo" />
    </header>
  )
}
