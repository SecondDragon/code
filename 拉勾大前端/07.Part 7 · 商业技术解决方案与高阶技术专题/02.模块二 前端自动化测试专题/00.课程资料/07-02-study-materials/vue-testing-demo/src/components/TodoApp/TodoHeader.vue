<template>
  <header class="header">
    <h1>todos</h1>
    <input
      class="new-todo"
      placeholder="What needs to be done?"
      autofocus
      data-testid="new-todo"
      @keyup.enter="handleNewTodo"
    />
  </header>
</template>

<script>
export default {
  name: 'TodoHeader',
  methods: {
    handleNewTodo (e) {
      const value = e.target.value.trim()
      if (!value.length) {
        return
      }
      this.$emit('new-todo', value)
      e.target.value = ''
    }
  }
}
</script>
