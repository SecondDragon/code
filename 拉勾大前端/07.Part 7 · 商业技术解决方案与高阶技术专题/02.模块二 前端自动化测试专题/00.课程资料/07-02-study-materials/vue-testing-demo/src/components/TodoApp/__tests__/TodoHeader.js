import { shallowMount } from '@vue/test-utils'
import TodoHeader from '@/components/TodoApp/TodoHeader.vue'

describe('TodoHeader.vue', () => {
  let wrapper = null
  beforeEach(() => {
    wrapper = shallowMount(TodoHeader)
  })
  test('New todo', async () => {
    const input = wrapper.find('input[data-testid="new-todo"]')
    const text = 'play'
    await input.setValue(text)
    await input.trigger('keyup.enter')
    expect(wrapper.emitted()['new-todo']).toBeTruthy()
    expect(wrapper.emitted()['new-todo'][0][0]).toBe(text)
    expect(input.element.value).toBe('')
  })

  test('snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
