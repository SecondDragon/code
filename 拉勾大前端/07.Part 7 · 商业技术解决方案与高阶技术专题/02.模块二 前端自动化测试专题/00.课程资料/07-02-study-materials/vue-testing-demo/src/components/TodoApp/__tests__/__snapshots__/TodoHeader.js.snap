// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoHeader.vue snapshot 1`] = `
<header class="header">
  <h1>todos</h1> <input placeholder="What needs to be done?" autofocus="autofocus" data-testid="new-todo" class="new-todo">
</header>
`;
